src/DynamoCoreWpf/Views/Preview/InfoBubbleView.xaml.cs

```diff
@@ -1023,39 +1024,79 @@ private void ScrollViewer_PreviewMouseWheel(object sender, MouseWheelEventArgs e
             e.Handled = true;
         }
 
-        private void DismissAllInfoButton_Click(object sender, RoutedEventArgs e)
+        /// <summary>
+        /// Retrieves all of a node's dismissed messages of a particular style.
+        /// </summary>
+        /// <param name="style"></param>
+        /// <returns></returns>
+        private ObservableCollection<InfoBubbleViewModel.InfoBubbleDataPacket> GetDismissedMessagesOfStyle(InfoBubbleViewModel.Style style)
+        {
+            return ViewModel.DismissedMessages.Where(x => x.Style == style).ToObservableCollection();
+        }
+
+        /// <summary>
+        /// Clears out the node's collection of all dismissed messages of a particular style (e.g. Error).
+        /// </summary>
+        /// <param name="style"></param>
+        private void ClearDismissedMessagesOfStyle(InfoBubbleViewModel.Style style)
         {
-            foreach (InfoBubbleDataPacket infoBubbleDataPacket in ViewModel.NodeMessages)
+            ObservableCollection<InfoBubbleViewModel.InfoBubbleDataPacket> messagesToDismiss = GetDismissedMessagesOfStyle(style);
+
+            for (int i = messagesToDismiss.Count - 1; i >= 0; i--)
             {
-                if (infoBubbleDataPacket.Style != InfoBubbleViewModel.Style.Info ||
-                    ViewModel.DismissedMessages.Contains(infoBubbleDataPacket))
-                {
-                    continue;
-                }
+                ViewModel.DismissedMessages.RemoveAt(i);
```

Comment:
Can you explain a bit more the syncing between `messagesToDismiss` and `DismissedMessages`?

Reply:
"Clears out the node's collection of all dismissed messages of a particular style (e.g. Error). When a user clicks on a particular error message to undismiss it, we undismiss all messages with that particular message string.

To cover the edge case where there are multiple messages with the same message string, but with different message styles (e.g. an Info Message, a Warning and an Error all with the same text), this method specifically un-dismisses the messages with both a matching style and message."